apheleia.el

```diff
@@ -389,18 +389,29 @@ as its sole argument."
                       ;; changes, and 2 if error.
                       (memq status '(0 1)))))))
 
-(defun apheleia--run-formatter (command callback)
-  "Run a code formatter on the current buffer.
-The formatter is specified by COMMAND, a list of strings or
-symbols (see `apheleia-format-buffer'). Invoke CALLBACK with one
-argument, a buffer containing the output of the formatter.
-
-If COMMAND uses the symbol `file' and the current buffer is
-modified from what is written to disk, then don't do anything."
+(defun apheleia--format-command (command &optional stdin)
+  "Format COMMAND into a shell-ccmd and list of file paths.
+Should return a list with the car being the optional input file-name
+the cadr being the optional output file-name and the cddr being the
+cmd to run.
+
+STDIN is the optional buffer to use when creating a temporary file for
+the formatters standard input.
+
+If COMMAND uses the symbol `file' and the current buffer is modified
+from what is written to disk, then return nil meaning meaning no
+cmd is to be run."
+  ;; WARN: The condition with regards to `file' only checks the original
+  ;; file so it doesn't apply the formats from multiple formatters being
+  ;; run. For example if formatter 1 produces some output and formatter
+  ;; 2 modifies file in place then the changes from formatter 1 may not
+  ;; apply at all.
   (cl-block nil
     (let ((input-fname nil)
           (output-fname nil)
           (npx nil))
+      ;; TODO: Support arbitrary package managers, not just npx.
+      ;; TODO: Should we do this only when npx is at the car of cmd?
```

Comment:
We're not actually calling into NPM here, this symbol is just used for setting `default-directory` from Elisp. The behavior could certainly be generalized, though, and probably should be.

The current behavior is that the same thing happens whether `npx` is at the beginning, or if it appears anywhere in the list. Unless there's a compelling reason otherwise, I'd suggest keeping that behavior for compatibility.